platform/core/src/services/UIModalService/index.js

```diff
@@ -34,20 +30,26 @@ function createUIModalService() {
 /**
  * Show a new UI modal;
  *
- * @param {Modal} component React component
- * @param {ModalProps} props { shouldCloseOnEsc, isOpen, closeButton, title, customClassName }
+ * @param {ModalProps} props { content, contentProps, shouldCloseOnEsc, isOpen, closeButton, title, customClassName }
  */
-function show(
-  component,
-  props = {
-    shouldCloseOnEsc: false,
-    isOpen: true,
-    closeButton: true,
-    title: null,
-    customClassName: null,
-  }
-) {
-  return uiModalServiceImplementation._show(component, props);
+function show({
+  content = null,
+  contentProps = null,
+  shouldCloseOnEsc = false,
+  isOpen = true,
+  closeButton = true,
+  title = null,
+  customClassName = null,
```

Comment:
Same here. I don't understand why `undefined` is a bad value. This implies that we're using `null` to indicate something a bit more intentional?

Reply:
There's nothing is waiting for null so it's fine to take it out, thanks.